lib/service/cfg.go

```diff
@@ -615,14 +615,65 @@ type Database struct {
 	StaticLabels map[string]string
 	// DynamicLabels is a list of database dynamic labels.
 	DynamicLabels services.CommandLabels
-	// CACert is an optional database CA certificate.
-	CACert []byte
+	// TLS keeps database connection TLS configuration.
+	TLS DatabaseTLS
 	// AWS contains AWS specific settings for RDS/Aurora/Redshift databases.
 	AWS DatabaseAWS
 	// GCP contains GCP specific settings for Cloud SQL databases.
 	GCP DatabaseGCP
 }
 
+// TLSMode defines all possible database verification modes.
+type TLSMode string
+
+const (
+	// VerifyFull is the strictest. Verifies certificate and server name.
+	VerifyFull TLSMode = "verify-full"
+	// VerifyCA checks the certificate, but is skips the server name verification.
+	VerifyCA TLSMode = "verify-ca"
+	// Insecure accepts any certificate.
+	Insecure TLSMode = "insecure"
+)
+
+// CheckAndSetDefaults check if TLSMode holds a correct value. If the value is not set
+// VerifyFull is set as a default. BadParameter error is returned if value set is incorrect.
+func (m *TLSMode) CheckAndSetDefaults() error {
+	switch *m {
+	case "": // Use VerifyFull if not set.
+		*m = VerifyFull
+	case VerifyFull, VerifyCA, Insecure:
+		// Correct value, do nothing.
+	default:
+		return trace.BadParameter("provided incorrect TLSMode value. Correct values are: %q %q %q",
+			VerifyFull, VerifyCA, Insecure)
```

Comment:
Nit: Usually in cases like this, we'd make a slice with all valid modes (e.g. `AllTLSModes = []string{VerifyFull, VerifyCA, Insecure}`) and use it here instead of enumerating each individually.

Reply:
Added